hoot-core/src/main/cpp/hoot/core/algorithms/WayJoiner2.cpp

```diff
@@ -54,34 +54,14 @@ namespace hoot
 
 HOOT_FACTORY_REGISTER(WayJoiner, WayJoiner2)
 
-WayJoiner2::WayJoiner2() :
-_leavePid(false),
-_numJoined(0)
+WayJoiner2::WayJoiner2()
 {
 }
 
 void WayJoiner2::join(const OsmMapPtr& map)
 {
-  _map = map;
-
-  //  Join back any ways with parent ids
-  _joinParentChild();
-  OsmMapWriterFactory::writeDebugMap(map, "after-way-joiner-join-parent-child-1");
-
-  //  Join any siblings that have the same parent id but the parent isn't connected
-  _joinSiblings();
-  OsmMapWriterFactory::writeDebugMap(map, "after-way-joiner-join-siblings");
-
-  //  Rejoin any ways that are now connected to their parents
-  _joinParentChild();
-  OsmMapWriterFactory::writeDebugMap(map, "after-way-joiner-join-parent-child-2");
-
-  //  Run one last join on ways that share a node and have a parent id
-  _joinAtNode();
-  OsmMapWriterFactory::writeDebugMap(map, "after-way-joiner-join-at-node");
-
-  //  Clear out any remaining unjoined parent ids
-  _resetParents();
+  //  Call base class implementation
+  WayJoiner::join(map);
```

Comment:
The only difference between these two implementations is what happens below, so consolidate that by calling the base class implementation.